<a href="https://colab.research.google.com/github/bwilli2086/data_analytics/blob/main/Python/Unit3_Python_Advanced_Pandas_DataWrangling_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment 1 : Python Data Analysis using salaries dataset

# Data Exploration:  This should include summary statistics, means, medians, quartiles, or any other relevant information about the data set. 

In [1]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/CunyLaguardiaDataAnalytics/datasets/master/Salaries.csv')

**1. Analyze the dataset**
1.Provide your observation and conclusions




In [3]:
type(df)

pandas.core.frame.DataFrame

In [4]:
df.shape

(397, 7)

In [5]:
df.head()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
0           1      Prof          B             19           18  Male  139750
1           2      Prof          B             20           16  Male  173200
2           3  AsstProf          B              4            3  Male   79750
3           4      Prof          B             45           39  Male  115000
4           5      Prof          B             40           41  Male  141500

In [6]:
df.tail()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
392         393      Prof          A             33           30  Male  103106
393         394      Prof          A             31           19  Male  150564
394         395      Prof          A             42           25  Male  101738
395         396      Prof          A             25           15  Male   95329
396         397  AsstProf          A              8            4  Male   81035

In [7]:
df.describe()

Unnamed: 0  yrs.since.phd  yrs.service         salary
count  397.000000     397.000000   397.000000     397.000000
mean   199.000000      22.314861    17.614610  113706.458438
std    114.748275      12.887003    13.006024   30289.038695
min      1.000000       1.000000     0.000000   57800.000000
25%    100.000000      12.000000     7.000000   91000.000000
50%    199.000000      21.000000    16.000000  107300.000000
75%    298.000000      32.000000    27.000000  134185.000000
max    397.000000      56.000000    60.000000  231545.000000

This file has a total of 397 rows and 7 columns. the name of the columns are: Unnamed: 0,	rank,	discipline,	yrs.since.phd,	yrs.service,	sex,	salary. The count shows that data was inputted for all columns on the 397 rows. average years since phd 22.314, avg years of service 17,614, average salary $113,706.46.

# Mean/median of all salaries


1.   Find the Mean and Median of all Salaries
2.  Provide your findings



In [27]:
# Mean of all salaries
df_mean=df[['salary']].mean()
df_mean

salary    113706.0
dtype: float64

The Mean of all Salaries is $113,706.46

In [29]:
# Median of all salaries
df_median=df[['salary']].median()
df_median

salary    107300.0
dtype: float64

Median of all salaries is $107,300.00

# Median salary between gender and/or rank(title) Etc...


Making a new Data Frame that only has the columns 'sex' and 'salary'

In [30]:
columns_to_be_drop=['Unnamed: 0','rank','discipline','yrs.since.phd','yrs.service']
type(columns_to_be_drop)

list

In [33]:
new_df=df.drop(columns_to_be_drop,axis=1)
new_df

sex  salary
0    Male  139750
1    Male  173200
2    Male   79750
3    Male  115000
4    Male  141500
..    ...     ...
392  Male  103106
393  Male  150564
394  Male  101738
395  Male   95329
396  Male   81035

[397 rows x 2 columns]

Finding the Median salary for all males

In [39]:
new_df
new_df.groupby('sex')[['salary']].median()

salary
sex           
Female  103750
Male    108043

Median salary for all males is $108,043.00

Finding the Median salary for all females

In [40]:
new_df
new_df.groupby('sex')[['salary']].median()

salary
sex           
Female  103750
Male    108043

Median salary for all females is $103,750.00

Comparing the Mean salary of Males and Females

In [45]:
new_df
new_df.groupby('sex')[['salary']].mean()

salary
sex                  
Female  101002.410256
Male    115090.418994

In [67]:
# The difference between mean salaries of male and female
new_df.groupby('sex').describe()

salary                               ...                               
        count           mean           std  ...       50%        75%       max
sex                                         ...                               
Female   39.0  101002.410256  25952.127317  ...  103750.0  117002.50  161101.0
Male    358.0  115090.418994  30436.927344  ...  108043.0  134863.75  231545.0

[2 rows x 8 columns]

# Data Wrangling – perform some basic transformations on the data

People who have 20 years of experience (Total: 157), (Mean Salary: 122,520.66)

In [59]:
df_filter_more=df[df['yrs.service'] >= 20]
df_filter_more


Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
3             4  Prof          B             45           39  Male  115000
4             5  Prof          B             40           41  Male  141500
6             7  Prof          B             30           23  Male  175000
7             8  Prof          B             45           45  Male  147765
8             9  Prof          B             21           20  Male  119250
..          ...   ...        ...            ...          ...   ...     ...
384         385  Prof          A             27           21  Male  125192
386         387  Prof          A             29           27  Male  139219
388         389  Prof          A             38           36  Male  119450
392         393  Prof          A             33           30  Male  103106
394         395  Prof          A             42           25  Male  101738

[157 rows x 7 columns]

In [56]:
df_filter_mean=df_filter[['salary']].mean()
df_filter_mean

salary    122520.656051
dtype: float64

Comparing the salaries of people who have 20+ years of experience vs less than 20 years of experience

In [70]:
df_filter_less=df[df['yrs.service'] < 20]
df_filter_less

Unnamed: 0       rank discipline  ...  yrs.service     sex  salary
0             1       Prof          B  ...           18    Male  139750
1             2       Prof          B  ...           16    Male  173200
2             3   AsstProf          B  ...            3    Male   79750
5             6  AssocProf          B  ...            6    Male   97000
9            10       Prof          B  ...           18  Female  129000
..          ...        ...        ...  ...          ...     ...     ...
390         391       Prof          A  ...           19    Male  166605
391         392       Prof          A  ...           19    Male  151292
393         394       Prof          A  ...           19    Male  150564
395         396       Prof          A  ...           15    Male   95329
396         397   AsstProf          A  ...            4    Male   81035

[240 rows x 7 columns]

KeyError: ignored

In [68]:
df_filter_more=df[df['yrs.service'] >= 20]
df_filter_more

Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
3             4  Prof          B             45           39  Male  115000
4             5  Prof          B             40           41  Male  141500
6             7  Prof          B             30           23  Male  175000
7             8  Prof          B             45           45  Male  147765
8             9  Prof          B             21           20  Male  119250
..          ...   ...        ...            ...          ...   ...     ...
384         385  Prof          A             27           21  Male  125192
386         387  Prof          A             29           27  Male  139219
388         389  Prof          A             38           36  Male  119450
392         393  Prof          A             33           30  Male  103106
394         395  Prof          A             42           25  Male  101738

[157 rows x 7 columns]

People who have salary of greater than or equal to 200,000

In [63]:
df_salary_more=df[df['salary'] >= 200000]
df_salary_more

Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
43           44  Prof          B             38           38  Male  231545
249         250  Prof          A             29            7  Male  204000
364         365  Prof          A             43           43  Male  205500

Your Findings here...


# Rename values, subset columns, etc

# Create some simple visualizations showing some simple statistics

Your observations from the visuals: 

# Draw some conclusions based on the data.  What did you discover?

Based on the analysis performed above, the conclusions are as follows:

Your Findings here...



I also discovered that the -

 Your Findings here...



